neuraxle/plotting.py

```diff
@@ -173,18 +176,22 @@ def on_next(self, value: Tuple[HyperparamsRepository, Trial]):
         :param value: hyperparams_repository, trial
         :return:
         """
+        # TODO : Rename function to something more meaningful like on_save_trial
         repo, trial = value
         if not self.plot_trial_on_next:
             return
 
         self._plot_all_trial_main_and_validation_metric_results(repo, trial)
 
     def _plot_all_trial_main_and_validation_metric_results(self, repo, trial):
-        trial_hash = repo._get_trial_hash(trial)
+        trial_hash = repo._get_trial_hash(trial.hyperparams.to_flat_dict())
```

Comment:
If it exists, we should should use the getter. I think there may be no getter here this time unfortunately.

Reply:
Well, I'd say its possible to retrieve it through the pipeline attribute of trial. That being said, we don't have a guaranteed that the pipeline attribute is set (e.g. it is None when we load a trial from a json file).